src/human_player_module.c

```diff
@@ -0,0 +1,391 @@
+ * 
+ * @param entered_value Value entered by the human player
+ * @return true - Entered value is right
+ * @return false - Value does not match the card values defined previously
+ */
+bool validate_card(char *entered_value)
+{
+    int size = sizeof CARD_VALUES / sizeof *CARD_VALUES;
+
+    for (int i = 0; i < size; ++i)
+    {
+        if (!strcmp(CARD_VALUES[i], entered_value))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * @brief Records the Human player input during his turn and performs appropriate functions 
+ * Called each time when it's the human player's turn 
+ * Keyboard Input from Human 
+ *        Q/q - Exit game
+ *        R/r - Requests card 
+ *        E/e - End Turn
+ *        Other- Map to appropriate Card Info
+ */
+
+ret_type_e record_human_input()
+{
+    char user_input[2];
+    printf("Please enter your choice \n");
+    scanf("%s", user_input);
+
+    if (user_input[0] == 'q' || user_input[0] == 'Q')
+    {
+        quit_game();
+    }
+    else if (user_input[0] == 'r' || user_input[0] == 'R')
+    {
+        request_card(HUMAN, 1);
+    }
+
+    else if (user_input[0] == 'e' || user_input[0] == 'E')
+    {
+        if (g_card_requested)
+        {
+            end_turn(g_player_on_turn);
+        }
+        else
+        {
+            printf("Please draw a card before you can end your turn");
+            display_player_deck(human_player);
+        }
+    }
+    else
+    {
+        if (validate_card(user_input))
+        {
+            //To be added if the format of user input is different
+            Card_t *human_card_choice = map_user_input(user_input);
+            //If valid add the card to the discard_pile
+            if (check_is_valid_turn(&g_card_on_table, human_card_choice))
+            {
+                Deck_t **pp_head = &(human_player->cards_on_hand);
+
+                Card_t *to_remove_card = pick_card_from_deck(pp_head, human_card_choice);
+
+                if (to_remove_card != NULL)
+                {
+                    RET_TYPE(remove_card_from_deck(pp_head, human_card_choice));
+                    add_card_at_end(g_discard_pile, *to_remove_card);
+                }
+            }
+        }
+        else
+        {
+            invalid_turn_warning();
+        }
+    }
+    return SUCCESS;
+}
+/**
+ * @brief Remove a particular card from the given deck
+ * Can be used to discard a particular card - during each turn
+ * Works for both the computer as well as human deck of cards
+ * 
+ * @param pp_head Pointer to the head of the deck from which card has to be removed
+ * @param card_to_be_removed  Card to be removed
+ * @return ret_type_e RET_EMPTY_LIST- If the deck has no cards
+ *                 RET_FAILURE - If the removing of card fails
+ *                 RET_SUCCESS - If the card is been removed successfully
+ */
+ret_type_e remove_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_removed)
```

Comment:
This function should be in cards_manegement.c instead of here.

Reply:
Yes, I have added a comment regarding the same in the function prototype.
Wanted to confirm with you and then make the changes

Reply:
Will be done in the next commit..